src/django_components/tag_formatter.py

```diff
@@ -0,0 +1,256 @@
+import abc
+import re
+from typing import List, NamedTuple
+
+from django.template import TemplateSyntaxError
+from django.utils.module_loading import import_string
+
+from django_components.app_settings import app_settings
+from django_components.expression import resolve_string
+from django_components.template_parser import VAR_CHARS
+from django_components.utils import is_str_wrapped_in_quotes
+
+TAG_RE = re.compile(r"^[{chars}]+$".format(chars=VAR_CHARS))
+
+
+class TagResult(NamedTuple):
+    """The return value from `TagFormatter.parse()`"""
+
+    component_name: str
+    """Component name extracted from the template tag"""
+    tokens: List[str]
+    """Remaining tokens (words) that were passed to the tag, with component name removed"""
+
+
+class TagFormatterABC(abc.ABC):
+    def start_block_tag(self, name: str) -> str:
+        """Formats the start tag of a block component."""
+        return self.start_tag(name)
+
+    def start_inline_tag(self, name: str) -> str:
+        """Formats the start tag of an inline component."""
+        return self.start_tag(name)
+
```

Comment:
Do we still need these?

Reply:
Yeah, no. When I was implementing this, my point of reference was a feature that would support syntaxes as found in django_components, django-web-components, and django-slippers. And django-slippers technically uses a different tag for inline and block tags. But since we're using the slash `/` to indicate inline component, this is not relevant anymore.

Update in https://github.com/EmilStenstrom/django-components/pull/572/commits/d9593ffdff0cd8dc087a2a04fff919e82680c432